vscode/src/edit/provider.ts

```diff
@@ -231,7 +237,7 @@ export class EditProvider {
 
     private async handleFileCreationResponse(text: string, isMessageInProgress: boolean): Promise<void> {
         const task = this.config.task
-        if (task.state !== CodyTaskState.pending) {
+        if (task.intent === 'test' && task.state !== CodyTaskState.working) {
```

Comment:
```suggestion
        if (task.intent === 'test' && task.state !== CodyTaskState.pending) {
```

I think we only want to create the file request if the task is at pending state since we only want to create the file once? It'd be cool to support multiple file creation though but i don't think that's something the test command needs right now?

Reply:
I just merged `CodyTaskState.pending` into `working` here, I think they're ultimately the same thing, we just need to add the extra check for `task.intent === 'test'` in our logic. Trying to make the Edit states a bit simpler!